src/table/components/withSelections.jsx

```diff
@@ -1,27 +1,20 @@
-import React, { useMemo } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import { selectCell } from '../utils/selections-utils';
+import { useContextSelector, TableContext } from '../context';
 import { getSelectionStyle } from '../utils/styling-utils';
+import { getCellSelectionState } from '../utils/selections-utils';
 
 export default function withSelections(CellComponent) {
   const HOC = (props) => {
-    const {
-      selectionState,
-      cell,
-      selectionDispatch,
-      styling,
-      themeBackgroundColor,
-      announce,
-      column,
-      ...passThroughProps
-    } = props;
-    const handleMouseUp = (evt) =>
-      cell.isDim && evt.button === 0 && selectCell({ selectionState, cell, selectionDispatch, evt, announce });
+    const { cell, styling, themeBackgroundColor, announce, column, ...passThroughProps } = props;
+    const cellSelectionState = useContextSelector(TableContext, (value) => getCellSelectionState(cell, value));
+    const selectionDispatch = useContextSelector(TableContext, (value) => value.selectionDispatch);
 
-    const selectionStyling = useMemo(
-      () => getSelectionStyle(styling, cell, selectionState, themeBackgroundColor),
-      [styling, cell, selectionState, themeBackgroundColor]
-    );
+    const handleMouseUp = (evt) => {
+      cell.isDim && evt.button === 0 && selectionDispatch({ type: 'select', payload: { cell, evt, announce } });
+    };
+
+    const selectionStyling = getSelectionStyle(styling, cellSelectionState, themeBackgroundColor);
```

Comment:
why not to use useMemo here?

Reply:
because it is low cost to run, only a switch returning the style. We shouldn't use useMemo if it doesn't give any value

Reply:
It might even be the case that the checks react needs to do for the dependency array (at least 2 checks) is more calculations than actually running `getSelectionStyling`

Reply:
oki